src/components/SearchbarComponent.tsx

```diff
@@ -11,93 +11,78 @@ export const Searchbar: React.FC = () => {
   return <FilteredList />;
 }
 
-var createReactClass = require("create-react-class");
-var FilteredList = createReactClass({
-  filterList: function (event: any) {
+const FilteredList: React.FC = () => {
+  const initNames: string[] = [];
+  const initButtons: React.Component[] = [];
+  const [buttons, setButtons] = React.useState(initButtons);
+  const [names, setNames] = React.useState(initNames);
+
+  function filterList(event: React.ChangeEvent<HTMLInputElement>) {
     searchBarText = event.target.value;
     console.log("The search bar val is " + searchBarText);
-    // Reset search bar when there's no input text
+    // Reset search bar whe there's no input text
     if (searchBarText == "") {
-      this.setState({ items: [] });
+      setButtons([]);
     } else {
-      var updatedList = this.state.initialItems;
-      updatedList = updatedList.filter(function (item: any) {
+      setInitialState();
+      var updatedList = names.filter(function (name: string) {
```

Comment:
Sorry I should have said this earlier but please try and use `const`. If it has to be dynamic then `let`. Try not to use var.